### DISCUSSION TITLE
Parametric Remote exported codes are always in raw

### DISCUSSION THREAD

**Discussion by bpsib** [bengtmartensson/IrScrutinizer#450] [Q&A] [answered]
I can't get IrScrutinizer to export remotes in parametric format. It converts to raw or pronto no matter what format is selected. I have `Lirc` selected but it exports the codes as raw as though I have selected `Lirc Raw`. Tested with a Samsung TV remote and I have the `Scrutinize Remote` tab and `Parametric Remote` subtab selected and rows are showing protocol as NECx2. How do I export codes as parametric?

**Comment by bengtmartensson** [comment]
Not sure if I understand your problem. Let there be something sensible in the parametric table, select Export pane, select either Girr or Text as export format, check Parameters, uncheck Raw and Pronto Hex, press Export param. remote. 

> I have Lirc selected but it exports the codes as raw as though I have selected Lirc Raw

Due to to limitations in the Lirc format, not all format (but I think more than half of them) can be expressed in Lirc "cooked" format. It will then fallback to the raw format.

**Comment by bpsib** [comment]
The problem I have is it converts to raw no matter what protocol I try including NEC. This is despite the rows showing them in parametric format. It should at least support NEC. What am I missing?

**Reply by bengtmartensson** [reply]
"NEC" is not a renderable protocol -- it normally an incomplete read and lacks information of the repeat. Please change it to e.g. NEC1.

**Reply by bpsib** [reply]
> "NEC" is not a renderable protocol -- it normally an incomplete read and lacks information of the repeat. Please change it to e.g. NEC1.

Same problem. Still raw codes.

**Reply by bengtmartensson** [reply]
I cannot follow you. You have to show with screenshots and/or a video sequence what you are trying to do.

**Comment by bengtmartensson** [comment]
Let me rephrase: Export of parametric signals works. The probability that you have discovered a new, substantial bug is essentially zero. So it is most likely a misunderstanding (or an unclear formulation in the documentation). If you tell me exactly what you want to achieve, you input data, what you get, and why you think that is wrong, I may be able to help.

**Comment by bpsib** [comment, marked as answer]
Apologies for the late reply. Here's what I doing with screenshots:

Export format is set to `Lirc`:
![Capture](https://user-images.githubusercontent.com/5899155/154860491-6cad1c40-d344-4410-af79-88f4cbbe84e6.PNG)

Scrutinize remote protocol set to NEC1 (originally tried NECx2):
![Capture2](https://user-images.githubusercontent.com/5899155/154860492-efdb8478-0a9b-451f-bc98-1bed161bb93d.PNG)

But exported signal codes are raw format regardlessly. Not sure if this is meant to be the case.
![Capture3](https://user-images.githubusercontent.com/5899155/154860494-0ace1b39-9966-4bcf-8c63-651c8108d66f.PNG)

What I ended up doing is using the spreadsheet export format and writing a BASH script to convert the decimal device function codes to Lirc hex format.

**Reply by bengtmartensson** [reply]
OK, now I see. It is indeed a bug (#453). Sorry. Thanx for finding it.

It has been fixed in the current sources. You can do any of these:

1. Change "NEC1" to "nec1",
2. Get [this file](https://raw.githubusercontent.com/bengtmartensson/IrScrutinizer/master/src/main/config/exportformats.d/lirc.xml) and drop it into your exportformats.d directory, overwriting the existing one,
3. get the [snapshot version](https://github.com/bengtmartensson/IrScrutinizer/releases/tag/ci-build)
